impl/src/main/kotlin/nl/nuts/consent/bridge/listener/CordaStateChangeListener.kt

```diff
@@ -134,6 +141,74 @@ class CordaStateChangeListener<S : ContractState>(
         // store in atomic reference, so that if the callback errors, the other thread can operate on it safely
         retryableStateUpdatesSubscription.set(subscription)
     }
+
+    private fun currentFeed(proxy: CordaRPCOps, asOfDateTime: Instant, stateClass: Class<S>): DataFeed<Vault.Page<S>, Vault.Update<S>> {
+        // feed criteria
+        return proxy.vaultTrackBy(
+            producedCriteria(asOfDateTime).or(consumedCriteria(asOfDateTime)),
+            PageSpecification(DEFAULT_PAGE_NUM, 100),
+            Sort(setOf(Sort.SortColumn(SortAttribute.Standard(Sort.CommonStateAttribute.STATE_REF), Sort.Direction.ASC))),
+            stateClass)
+    }
+
+    private fun publishMissedProducedStates(proxy: CordaRPCOps, asOfDateTime: Instant, clazz: Class<S>) {
+        var historyDone = false
+        var tempFeed: Vault.Page<S>? = null
+        while (!historyDone) {
+            tempFeed = proxy.vaultQueryBy(
+                producedCriteria(asOfDateTime),
+                PageSpecification(DEFAULT_PAGE_NUM, 100),
+                Sort(setOf(Sort.SortColumn(SortAttribute.Standard(Sort.CommonStateAttribute.STATE_REF), Sort.Direction.ASC))),
+                clazz)
+
+            // update exit criteria
+            historyDone = tempFeed.states.size < 100
```

Comment:
magic number and error prone, I'd think it should match the '100' just above? Maybe make it a constant?